tests/DataCache.test.ts

```diff
@@ -131,4 +131,88 @@ describe("DataCache", () => {
     const cache = new DataCache<DisposableString>(3);
     expect(() => cache.insert("1", new DisposableString("A"), 4)).toThrowError(ANY_ERROR);
   });
+
+  it("reserves keys and prevents them from being evicted", () => {
+    const cache = new DataCache<DisposableString>(3);
+    cache.insert("1", new DisposableString("A"));
+    cache.insert("2", new DisposableString("B"));
+    cache.insert("3", new DisposableString("C"));
+    cache.setReservedKeys(new Set(["1", "2"]));
+    expect(cache.size).toBe(3);
+
+    // 3 should be evicted because 1 and 2 are reserved.
+    cache.insert("4", new DisposableString("D"));
+    expect(cache.size).toBe(3);
+    expect(cache.get("1")?.value).toBe("A");
+    expect(cache.get("2")?.value).toBe("B");
+    expect(cache.get("3")?.value).toBe(undefined);
+    expect(cache.get("4")?.value).toBe("D");
+  });
+
+  it("allows unreserved keys to be evicted", () => {
+    const cache = new DataCache<DisposableString>(2);
+    cache.insert("1", new DisposableString("A"));
+    cache.insert("2", new DisposableString("B"));
+    cache.setReservedKeys(new Set(["1"]));
+    expect(cache.size).toBe(2);
+
+    cache.insert("3", new DisposableString("C"));
+    expect(cache.size).toBe(2);
+    // 2 is evicted
+    expect(cache.get("1")?.value).toBe("A");
+    expect(cache.get("2")?.value).toBe(undefined);
+    expect(cache.get("3")?.value).toBe("C");
+
+    // Remove A from the reserved keys. It will be added to the front of the list.
```

Comment:
nit: you're sometimes referring to the key and sometimes the value as the thing. Ie, up above you say "2 is evicted" but here you refer to "Remove A..."